src/java.desktop/macosx/classes/sun/lwawt/macosx/CAccessibility.java

```diff
@@ -469,6 +469,9 @@ public Boolean call() throws Exception {
     public static Accessible accessibilityHitTest(final Container parent, final float hitPointX, final float hitPointY) {
         return invokeAndWait(new Callable<Accessible>() {
             public Accessible call() throws Exception {
+                if (parent == null || !parent.isVisible()) {
```

Comment:
Should the `isShowing` be used here to check containers up to the top level? But I think it should be possible to create a test that may trigger an exception in `getLocationOnScreen()` after the check, so maybe try/catch will be a better approach.

Reply:
Well, if the parent is visible then it is all we need to get its screen coordinates. But i also think that surrounding the entire call in try/catch block would work better. Updating the fix.

Reply:
Why do you need to wrap the entire call, can this exception be thrown by some other methods?

Reply:
I think it is better to wrap only one method, it will be obvious where the exception can be thrown and it does not "fill the git history" of the method by all these unnecessary shifting.

Reply:
Ok, reverting to only do that one call.